pulse-android-otel/pulse-upload-plugin/src/main/kotlin/com/pulse/upload/PulseUploadSourceMapsPlugin.kt

```diff
@@ -0,0 +1,335 @@
+package com.pulse.upload
+
+import java.io.File
+import java.net.HttpURLConnection
+import java.net.URL
+import org.gradle.api.DefaultTask
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.options.Option
+import org.gradle.api.tasks.TaskAction
+
+class PulseUploadSourceMapsPlugin : Plugin<Project> {
+    override fun apply(project: Project) {
+        project.tasks.register("uploadSourceMaps", UploadSourceMapsTask::class.java) {
+            group = "pulse"
+            description = "Upload ProGuard/R8 mapping files to Pulse backend"
+            configFile.convention(project.objects.fileProperty())
+            projectDirectory.set(project.layout.projectDirectory.asFile.absolutePath)
+        }
+    }
+}
+
+abstract class UploadSourceMapsTask : DefaultTask() {
+
+    @get:Option(option = "api-url", description = "Backend API URL for uploading source maps")
+    @get:Input
+    abstract val apiUrl: Property<String>
+
+    @get:Option(option = "mapping-file", description = "ProGuard/R8 mapping file path")
+    @get:Input
+    abstract val mappingFileString: Property<String>
+
+    @get:Internal
+    abstract val configFile: RegularFileProperty
+
+    @get:Option(option = "app-version", description = "App version (e.g., 1.0.0). Required.")
+    @get:Input
+    abstract val appVersion: Property<String>
+
+    @get:Option(option = "version-code", description = "Version code (positive integer, e.g., 1). Required.")
+    @get:Input
+    abstract val versionCode: Property<String>
+
+    @get:Option(option = "verbose", description = "Show verbose/debug information")
+    @get:Input
+    abstract val verbose: Property<Boolean>
+
+    @get:Input
+    abstract val projectDirectory: Property<String>
```

Comment:
```suggestion
    @get:Input
    internal abstract val projectDirectory: Property<String>
```
as this you can set from the plugin itself. You don't have to expose this to the end users